src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -78,17 +79,32 @@ protected override void BeginInvokeJS(long asyncHandle, string identifier, strin
         {
             if (_clientProxy is null)
             {
-                throw new InvalidOperationException(
-                    "JavaScript interop calls cannot be issued at this time. This is because the component is being " +
-                    $"statically rendered. When prerendering is enabled, JavaScript interop calls can only be performed " +
-                    $"during the OnAfterRenderAsync lifecycle method.");
+                if (_permanentlyDisconnected)
+                {
+                    throw new JSDisconnectedException(
+                           "JavaScript interop calls cannot be issued at this time. This is because the circuit has disconnected " +
+                           "and is being disposed.");
+                }
+                else
```

Comment:
Minor: Do we have a convention to have `else` always? I would simply not have it and throw.

Reply:
In cases where the `if`/`else` clauses could be written equally in either order (like this one), I usually prefer to have the `else` explicitly so as to make the two halves of more equal status to each other. I think it more closely matches how people speak in natural language when talking about two alternatives. For example a person might say "*if it's sunny go to the park, otherwise go to the museum*", whereas they wouldn't say "*if it's sunny go to the park and ignore anything else I say. Go to the museum.*"

I normally prefer to skip the `else` in cases where the `if` part plays the role of a guard clause, where it's natural to terminate the flow if the guard clause applies. This is more like "*Customers under 18 are not permitted. All customers must wear a mask.*"

Of course it's all subjective and there isn't really a ground truth here. Based on preference I propose to leave it, but if this is really troubling to anyone, I'll change it!